core/src/main/java/com/google/cloud/sql/core/ConnectionInfo.java

```diff
@@ -51,4 +52,29 @@ Map<IpType, String> getIpAddrs() {
   SslData getSslData() {
     return sslData;
   }
+
+  ConnectionMetadata toConnectionMetadata(
```

Comment:
Moved here because `BaseConnectionInfoCache` was deleted. 

(Additional context: `ConnectionMetadata` is part of the legacy public Java API contract.)

Reply:
This seems like the perfect place for such concerns -- basically the connection info data class.